models/issue.go

```diff
@@ -962,6 +990,179 @@ func Issues(opts *IssuesOptions) ([]*Issue, error) {
 	return issues, nil
 }
 
+// numericQuery an numeric-equality query for the given value and field
+func numericQuery(value int64, field string) *query.NumericRangeQuery {
+	f := float64(value)
+	tru := true
+	q := bleve.NewNumericRangeInclusiveQuery(&f, &f, &tru, &tru)
+	q.SetField(field)
+	return q
+}
+
+// SearchIssues searches for issues by given conditions.
+func SearchIssues(opts *IssuesOptions) ([]*Issue, error) {
+	subqueries := []query.Query{
+		numericQuery(opts.RepoID, "RepoID"),
+	}
+
+	if len(opts.Keyword) > 0 {
+		fields := strings.Fields(strings.ToLower(opts.Keyword))
+		subqueries = append(subqueries, bleve.NewDisjunctionQuery(
+			bleve.NewPhraseQuery(fields, "Title"),
+			bleve.NewPhraseQuery(fields, "Content"),
+		))
+	}
+
+	switch opts.IsClosed {
+	case util.OptionalBoolTrue:
+		isClosedQuery := bleve.NewBoolFieldQuery(true)
+		isClosedQuery.SetField("IsClosed")
+		subqueries = append(subqueries, isClosedQuery)
+	case util.OptionalBoolFalse:
+		isClosedQuery := bleve.NewBoolFieldQuery(false)
+		isClosedQuery.SetField("IsClosed")
+		subqueries = append(subqueries, isClosedQuery)
+	}
+
+	switch opts.IsPull {
+	case util.OptionalBoolTrue:
+		isPullQuery := bleve.NewBoolFieldQuery(true)
+		isPullQuery.SetField("IsPull")
+		subqueries = append(subqueries, isPullQuery)
+	case util.OptionalBoolFalse:
+		isPullQuery := bleve.NewBoolFieldQuery(false)
+		isPullQuery.SetField("IsPull")
+		subqueries = append(subqueries, isPullQuery)
+	}
+
+	if opts.MilestoneID > 0 {
+		subqueries = append(subqueries, numericQuery(opts.MilestoneID, "MilestoneID"))
+	}
+
+	if opts.AssigneeID > 0 {
+		subqueries = append(subqueries, numericQuery(opts.AssigneeID, "AssigneeID"))
+	}
+	if opts.PosterID > 0 {
+		subqueries = append(subqueries, numericQuery(opts.PosterID, "PosterID"))
+	}
+	if opts.MentionedID > 0 {
+		subqueries = append(subqueries,
+			bleve.NewPhraseQuery([]string{strconv.FormatInt(opts.MentionedID, 36)},
+				"Mentions"))
+	}
+
+	if len(opts.Labels) > 0 && opts.Labels != "0" {
+		labelIDs, err := base.StringsToInt64s(strings.Split(opts.Labels, ","))
+		if err != nil {
+			log.Warn("Invalid Labels argument: %s", opts.Labels)
+		} else {
+			for _, labelID := range labelIDs {
+				subqueries = append(subqueries,
+					bleve.NewPhraseQuery([]string{strconv.FormatInt(labelID, 36)},
+						"Labels"))
+			}
+		}
+	}
+
+	query := bleve.NewConjunctionQuery(subqueries...)
+	var search *bleve.SearchRequest
+	if opts.Page >= 0 {
+		var startFrom int
+		if opts.Page == 0 {
+			startFrom = 0
+		} else {
+			startFrom = (opts.Page - 1) * setting.UI.IssuePagingNum
+		}
+		search = bleve.NewSearchRequestOptions(query, setting.UI.IssuePagingNum, startFrom, false)
+	} else {
+		search = bleve.NewSearchRequestOptions(query, 2147483647, 0, false)
+	}
+
+	search.Fields = []string{
+		"ID",
```

Comment:
I'm a little confused: in https://github.com/go-gitea/gitea/issues/379#issuecomment-267489391 you said "If we use bleve, SQL query will be no need". I thought that meant that all of the issue columns should be in the indexer. @tboerger's comment (https://github.com/go-gitea/gitea/pull/387#issuecomment-267621751) also says to add all of the columns to the indexer.

I personally prefer the approach of only storing the title and content in the indexer, and filtering by and loading other columns from the database, and would be happy to change my PR to use this approach. I just want to make sure that I'm understanding you before moving forward.